src/sensor/sensor.h

```diff
@@ -94,11 +95,27 @@ class Sensor : public QObject
      */
     QThread* detectorThread() { return &_detectorThread; };
 
+
+    /**
+     * @brief Return flasher class used by this sensor
+     *  TODO: This should be moved to a singleton flasher instance
```

Comment:
What is the reason it should be a singleon? What if there is more than one device connected or more than one type of device connected?

Reply:
Because flasher is a service as the others singletons that we have,
I dont see the relation of multiple devices and the flasher class as a singleton.

Reply:
What if you want to flash two devices at once? What if you want to flash an avr or microchip instead of stm?

Reply:
> What if you want to flash two devices at once?

This will not work with multiple flasher instances, since stm32flash binary or the serial port can be locked by others instances. 

> What if you want to flash an avr or microchip instead of stm?

That's why we have the flasher class, to abstract the flash method for each hardware/solution.